mapper/pojo-base/src/main/java/org/hibernate/search/mapper/pojo/mapping/definition/annotation/processing/impl/AbstractFieldAnnotationProcessor.java

```diff
@@ -42,12 +46,21 @@ public final void process(PropertyMappingStep mappingContext, A annotation,
 		PropertyMappingFieldOptionsStep<?> fieldContext =
 				initFieldMappingContext( mappingContext, annotation, cleanedUpRelativeFieldName );
 
+		ValueBinderRef valueBinder = getValueBinder( annotation );
 		ValueBinder binder = createValueBinder(
 				getValueBridge( annotation ),
-				getValueBinder( annotation ),
+				valueBinder,
 				context
 		);
-		fieldContext.valueBinder( binder );
+
+		if ( valueBinder.params() != null ) {
+			Map<String, Object> params = Arrays.stream( valueBinder.params() )
+					.collect( Collectors.toMap( Parameter::name, Parameter::value ) );
```

Comment:
I'd suggest an implementation similar to `org.hibernate.search.cfg.spi.ParameterAnnotationsReader#toNewMutableMap` in Hibernate Search 5 instead. It handles duplicate parameter names more cleanly.

Also, keep in mind you can factorize that implementation by exposing it in `org.hibernate.search.mapper.pojo.mapping.definition.annotation.processing.MappingAnnotationProcessorContext` with a `Map<String, String> toMap(Parameter[] params)` method. It's API, but I assume this can be useful to users, too.